compiler/rustc_borrowck/src/universal_regions.rs

```diff
@@ -241,15 +243,15 @@ impl<'tcx> UniversalRegions<'tcx> {
         tcx: TyCtxt<'tcx>,
         closure_substs: SubstsRef<'tcx>,
         expected_num_vars: usize,
-        typeck_root_def_id: DefId,
+        closure_def_id: LocalDefId,
     ) -> IndexVec<RegionVid, ty::Region<'tcx>> {
         let mut region_mapping = IndexVec::with_capacity(expected_num_vars);
         region_mapping.push(tcx.lifetimes.re_static);
         tcx.for_each_free_region(&closure_substs, |fr| {
             region_mapping.push(fr);
         });
 
-        for_each_late_bound_region_defined_on(tcx, typeck_root_def_id, |r| {
+        for_each_late_bound_region_in_recursive_scope(tcx, tcx.local_parent(closure_def_id), |r| {
```

Comment:
Nit: This functions now assumes being called from the strict parent of the closure, not any higher parent. Can we make it a method on UniversalRegions and add `assert_eq!(self.len(), expected_num_vars)` and `assert_eq!(self.defining_ty.def_id(), tcx.parent(closure_def_id)`

Reply:
> Can we make it a method on UniversalRegions

Make what a method? `for_each_late_bound_region_in_recursive_scope`?

> and add assert_eq!(self.len(), expected_num_vars)

What's the difference between this assertion and the one that happens below?

Reply:
> Make what a method?

I meant `closure_mapping`.

> What's the difference between this assertion and the one that happens below?

Oops, I missed that. I guess it is sufficient then.